### PR TITLE
fix(card): wrong props on not-interactive card - FE-4243

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
Removing prop tabIndex and type when card is not interactive
 
fixes: #4262 

### Current behaviour
Props tabIndex and type are provided even when card is non interactive. It causes problem becase it allows users to tab into the card but there is no visual representation

### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] <del> Storybook added or updated if required
- [ ] <del> Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### Testing instructions
Testable in chromatic storybook => Card => default

### CODE REVIEW

**Review by grabkowski — COMMENTED** [review]
I think that the message should be:
`fix(card): incorrect props on non-interactive card`

or just:
`fix(card): incorrect props on a static card`

**Review by DipperTheDan — COMMENTED** [review]
Code looks OK. I agree with @grabkowski that the commit message should be one of the two that has been suggested in the comment. Fix the commit message and I'll gladly approve 😃

File: src/components/card/card.component.js [hunk 1]
```diff
@@ -57,17 +57,21 @@ const Card = ({
 
   const onClickHandler = interactive ? handleClick : null;
 
+  const interactiveProps = interactive && {
+    tabIndex: 0,
+    type: "button",
+  };
+
```
Comment (mmaciorowski-sage) [inline src/components/card/card.component.js:64]: I'm not sure if it matters in this case but ternary is more clear for me
Zrzut ekranu 2021-08-2 o 10 46 49

**Review by mmaciorowski-sage — DISMISSED** [review]

File: src/components/card/card.component.js (hunk 1 above)
Comment (dawidzarzycki) [reply src/components/card/card.component.js:64]: Thank you for your suggestion, anyway I would leave it as it is now. We don't need the `undefined` as we don't want to do anything with it. All we need is just `tabIndex` and `type` prop if the card is interactive. I think this is clear enough. We use this approach in our project very often

Comment (dawidzarzycki) [reply src/components/card/card.component.js:64]: Thank you for your suggestion. If the interactive will be `false` or `undefined` it won't change anything as none of these props will be shown in DOM. Anyway I think that undefined approach is cool. I'll change it

**Review by DipperTheDan — APPROVED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**